Main.cpp

```diff
@@ -100,9 +101,10 @@ void ImageMapsInDirectoryFromConsole(const string& directory, RenderSettings ren
 	
 	filenames.insert(std::end(filenames), std::begin(saveFilenames), std::end(saveFilenames));
 
-	for (size_t i = filenames.size() - 1; i >= 0; i--)
+	// Loop starts at index size - 1 and ends after index 0 executes
+	for (std::size_t i = filenames.size(); i-- > 0; )
 	{
-		string filename = XFile::GetFilename(filenames[i]);
+		string& filename = XFile::GetFilename(filenames[i]);
```

Comment:
I see an error for this line on Linux. I believe it needs to be `const`. The error message is:
```
Main.cpp:107:11: error: non-const lvalue reference to type 'basic_string<...>'
      cannot bind to a temporary of type 'basic_string<...>'
                string& filename = XFile::GetFilename(filenames[i]);
                        ^          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
MSVC doesn't mind the code in this state. Makes more sense to make it const though looking at it again and it does need to compile on Linux I suppose. :)

Reply:
Yes, I think MSVC is more forgiving on this one. It does of course work. It's just the standard explicitly calls it out as bad practice and disallows it, since updates to a temporary object are probably a bug.